blazor/gantt-chart/scrolling.md

```diff
@@ -307,4 +307,305 @@ RowIndex : <input @bind-value = "@RowIndex" />
+    <GanttColumns>
+        <GanttColumn Field="TaskId" IsFrozen=true></GanttColumn>
+        <GanttColumn Field="TaskName" IsFrozen=true></GanttColumn>
+        <GanttColumn Field="StartDate"></GanttColumn>
+        <GanttColumn Field="EndDate"></GanttColumn>
+        <GanttColumn Field="Duration"></GanttColumn>
+        <GanttColumn Field="Progress"></GanttColumn>
+    </GanttColumns>
+    <GanttSplitterSettings Position="70%"></GanttSplitterSettings>
+</SfGantt>
+
+@code {
+    private List<TaskData> TaskCollection { get; set; }
+    protected override void OnInitialized()
+    {
+        this.TaskCollection = GetTaskCollection();
+    }
+
+    public class TaskData
+    {
+        public int TaskId { get; set; }
+        public string TaskName { get; set; }
+        public DateTime StartDate { get; set; }
+        public DateTime EndDate { get; set; }
+        public string Duration { get; set; }
+        public int Progress { get; set; }
+        public int? ParentId { get; set; }
+    }
+
+    public static List<TaskData> GetTaskCollection()
+    {
+        List<TaskData> Tasks = new List<TaskData>() {
+        new TaskData() { TaskId = 1, TaskName = "Project initiation", StartDate = new DateTime(2023, 02, 05), EndDate = new DateTime(2022, 02, 21) },
+        new TaskData() { TaskId = 2, TaskName = "Identify Site location", StartDate = new DateTime(2023, 02, 05), Duration = "0", Progress = 30, ParentId = 1 },
+        new TaskData() { TaskId = 3, TaskName = "Perform soil test", StartDate = new DateTime(2023, 02, 05), Duration = "4", Progress = 40, ParentId = 1 },
+        new TaskData() { TaskId = 4, TaskName = "Soil test approval", StartDate = new DateTime(2023, 02, 05), Duration = "0", Progress = 30, ParentId = 1 },
+        new TaskData() { TaskId = 5, TaskName = "Project estimation", StartDate = new DateTime(2023, 02, 06), EndDate = new DateTime(2023, 02, 21) },
+        new TaskData() { TaskId = 6, TaskName = "Develop floor plan for estimation", StartDate = new DateTime(2023, 02, 06), Duration = "3", Progress = 30, ParentId = 5 },
+        new TaskData() { TaskId = 7, TaskName = "List materials", StartDate = new DateTime(2023, 02, 06), Duration = "3", Progress = 40, ParentId = 5 },
+        new TaskData() { TaskId = 8, TaskName = "Estimation approval", StartDate = new DateTime(2023, 02, 06), Duration = "0", Progress = 30, ParentId = 5 }
+    };
+        return Tasks;
+    }
+}
+```
+
+![Blazor Gantt Chart with Frozen Columns](./images/blazor_Gantt_chart_particular_column_frozen.gif)
+
+### Freeze direction
+
+You can freeze Gantt columns on the left or right side by using the [Freeze](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Gantt.GanttColumn.html#Syncfusion_Blazor_Gantt_GanttColumn_Freeze) property, while the remaining columns remain movable. The Gantt chart will automatically adjust the column positions based on the Freeze value. Additionally, the `IsFrozen` property should be set to true for the frozen columns.
```

Comment:
The Freeze property in the Gantt column allows you to freeze the column either the left or right side of the grid while the remaining columns remain movable. The IsFrozen property should be set to true for the frozen columns. 

The FreezeDirection property has two values: FreezeDirection.Left allows you to freeze columns on the left side, while FreezeDirection.Right allows you to freeze columns on the right side.

To use FreezeDirection in the Blazor Gantt chart, you need to set IsFrozen property of GanttColumn in addition to the Freeze property. In the below code sample , the TaskId column is frozen on the left side of the grid, while the TaskName column is frozen on the right side.

The following is the sample code for using FreezeDirection in the Blazor Gantt chart:

Reply:
Content changed